internal/scorecard/alpha/run_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alpha
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestFakeRunner(t *testing.T) {
+
+	cases := []struct {
+		name            string
+		configPathValue string
+		selector        string
+		wantError       bool
+	}{
+		{
+			name:            "should execute 1 fake test",
+			configPathValue: "testdata/bundle",
+			selector:        "suite=basic",
+			wantError:       false,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.configPathValue, func(t *testing.T) {
+			o := Scorecard{}
+			var err error
+			o.Config, err = LoadConfig(c.configPathValue)
+			if err != nil {
+				t.Fatalf("Unexpected error loading config %w", err)
+			}
+			o.Selector, err = labels.Parse(c.selector)
+			if err != nil {
+				t.Fatalf("Unexpected error parsing selector %w", err)
+			}
+			o.SkipCleanup = true
+
+			mockResult := v1alpha2.ScorecardTestResult{}
+			mockResult.Name = "mocked test"
+			mockResult.Description = "mocked test description"
+			mockResult.State = v1alpha2.PassState
+			mockResult.Errors = make([]string, 0)
+			mockResult.Suggestions = make([]string, 0)
+
+			runner := FakePodTestRunner{}
+			runner.TestConfig = o
+			runner.TestResult = &mockResult
+			o.TestRunner = runner
+
+			ctx, cancel := context.WithTimeout(context.Background(), time.Duration(7*time.Second))
+			defer cancel()
+			_, err = o.RunTests(ctx)
+
+			if err == nil && c.wantError {
+				t.Fatalf("Wanted error but got no error")
+			} else if err != nil {
+				if !c.wantError {
+					t.Fatalf("Wanted result but got error: %v", err)
+				}
+				return
+			}
+
+		})
+
+	}
+}
+
+func TestFakeRunnerWithTestError(t *testing.T) {
```

Comment:
Make this just another case in `TestRunTests` and have `testRunner` be a field on the test case?

Reply:
done.